botocore/client.py

```diff
@@ -208,6 +209,10 @@ def _get_client_args(self, service_model, region_name, is_secure,
         self._inject_s3_configuration(
             config_kwargs, scoped_config, client_config)
         self._conditionally_unregister_fix_s3_host(endpoint_url, event_emitter)
+        if service_name == 's3':
+            # We want to make double sure that we don't create this object
```

Comment:
We already limit it to S3 by how we register the events. I also do not think it would be a significant memory imprint. I just find it ugly when reading the code to have a big if statement for a specific service in the middle of generic logic. @jamesls want to be the tie breaker?